GPCS4/SceModules/SceFiber/sce_fiber.cpp

```diff
@@ -32,25 +64,52 @@ int PS4API sceFiberReturnToThread(void)
 }
 
 
-int PS4API sceFiberRun(void)
+int PS4API sceFiberRun(SceFiber* fiber, uint64_t argOnRunTo, uint64_t* argOnReturn)
 {
 	LOG_FIXME("Not implemented");
+	fiber->fiber = UtilFiber::CreateAndRunFiber(fiber->entry, fiber->argOnInitialize, argOnRunTo);
+	currentFiber = fiber;
 	return SCE_OK;
 }
 
 
-int PS4API sceFiberSwitch(void)
+int PS4API sceFiberSwitch(SceFiber* fiber, uint64_t argOnRunTo, uint64_t* argOnRun)
 {
 	LOG_FIXME("Not implemented");
+
+	if (fiber->fiber == NULL)
+	{
+		fiber->fiber = UtilFiber::CreateAndRunFiber(fiber->entry, fiber->argOnInitialize, argOnRunTo);
+	}
+	else
+	{
+		UtilFiber::SwitchToFiber(fiber->fiber);
+	}
+
+	currentFiber = fiber;
+	
 	return SCE_OK;
 }
 
-int PS4API sceFiberGetSelf(void)
+
+int PS4API sceFiberGetSelf(SceFiber** fiber)
 {
 	LOG_FIXME("Not implemented");
+	*fiber = currentFiber;
```

Comment:
can we implement this using `GetCurrentFiber`

Reply:
GetCurrentFiber returns Windows Fiber, but we need SceFiber. Probably we need vector of created sceFibers and search current windows fiber in this vector

Reply:
No, there's no difference by using a global `vector` compared to a global variable.
We can set the `SceFiber` in the fiber's local storage during `CreateFiber` by passing `lpParameter`, then use `GetFiberData` to retrieve the data.